Docker build command is not working anymore

Since the merge with the TokyoFarmer fork the command `docker build . ` is not working anymore.

 Reeding the commit history, it was introduced a change that uses "go 1.11 modules support".

The Docker file is installing the default ubuntu xenial version of Go that is 1.6.2.

Whit this changes to the RUN command in the Docker file I was able to install the rigth version:
```
RUN \
	apt-get update && \
	DEBIAN_FRONTEND=noninteractive \
	    apt-get upgrade -y && \
		apt-get install -y \
		        locales \
			unoconv \
			gcc \
			supervisor \
			software-properties-common &&\
			add-apt-repository ppa:longsleep/golang-backports &&\
			apt-get update -y &&\
			apt-get install -y golang-go &&\
		go install github.com/HeavyHorst/unoconv-api && \
        apt-get remove -y golang-go gcc && \
	    apt-get autoremove -y && \
        apt-get clean && \
	    rm -rf /var/lib/apt/lists/
```

At this point I started to get a new error when the build try to execute `go install github.com/HeavyHorst/unoconv-api`.

```
go/src/github.com/HeavyHorst/unoconv-api/main.go:14:2: cannot find package "github.com/okzk/sdnotify" in any of:
        /usr/lib/go-1.11/src/github.com/okzk/sdnotify (from $GOROOT)
        /go/src/github.com/okzk/sdnotify (from $GOPATH)
go/src/github.com/HeavyHorst/unoconv-api/main.go:10:2: cannot find package "github.com/rs/xaccess" in any of:
        /usr/lib/go-1.11/src/github.com/rs/xaccess (from $GOROOT)
        /go/src/github.com/rs/xaccess (from $GOPATH)
go/src/github.com/HeavyHorst/unoconv-api/main.go:11:2: cannot find package "github.com/rs/xhandler" in any of:
        /usr/lib/go-1.11/src/github.com/rs/xhandler (from $GOROOT)
        /go/src/github.com/rs/xhandler (from $GOPATH)
go/src/github.com/HeavyHorst/unoconv-api/config.go:6:2: cannot find package "github.com/rs/xlog" in any of:
        /usr/lib/go-1.11/src/github.com/rs/xlog (from $GOROOT)
        /go/src/github.com/rs/xlog (from $GOPATH)
go/src/github.com/HeavyHorst/unoconv-api/handler.go:12:2: cannot find package "github.com/rs/xmux" in any of:
        /usr/lib/go-1.11/src/github.com/rs/xmux (from $GOROOT)
        /go/src/github.com/rs/xmux (from $GOPATH)
go/src/github.com/HeavyHorst/unoconv-api/utils.go:7:2: cannot find package "github.com/saintfish/chardet" in any of:
        /usr/lib/go-1.11/src/github.com/saintfish/chardet (from $GOROOT)
        /go/src/github.com/saintfish/chardet (from $GOPATH)
go/src/github.com/HeavyHorst/unoconv-api/handler.go:13:2: cannot find package "golang.org/x/net/context" in any of:
        /usr/lib/go-1.11/src/golang.org/x/net/context (from $GOROOT)
        /go/src/golang.org/x/net/context (from $GOPATH)
go/src/github.com/HeavyHorst/unoconv-api/utils.go:6:2: cannot find package "gopkg.in/iconv.v1" in any of:
        /usr/lib/go-1.11/src/gopkg.in/iconv.v1 (from $GOROOT)
        /go/src/gopkg.in/iconv.v1 (from $GOPATH)
```

I have almost no experience with golang. Apparently he is not using the go.mod file correctly.

I have fixed this using a docker multi-stage build. Would a PR be welcome?

--- a/Dockerfile
+++ b/Dockerfile
@@ -35,5 +35,8 @@ COPY supervisord.conf /etc/supervisor/conf.d/supervisord.conf
 # Expose 3000
 EXPOSE 3000
 
+HEALTHCHECK --interval=30s --timeout=10s \
+    CMD curl http://localhost:3000/unoconv/health
+
 # Startup
 ENTRYPOINT ["/usr/bin/supervisord"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,12 +1,14 @@
+FROM golang:1.11 as api-builder
+WORKDIR /unoconv-api
+COPY . /unoconv-api
+RUN go build
+
+
 FROM ubuntu:xenial
 
-MAINTAINER Rene Kaufmann <kaufmann.r@gmail.com>
+LABEL maintainer="kaufmann.r@gmail.com"
 
-ENV GOPATH /go
-ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
-ENV GO15VENDOREXPERIMENT 1
-
-ADD . /go/src/github.com/HeavyHorst/unoconv-api
+COPY --from=api-builder /unoconv-api/unoconv-api /opt/unoconv-api/unoconv-api
 
 #Install unoconv
 RUN \
@@ -16,14 +18,11 @@ RUN \
 		apt-get install -y \
 		        locales \
 			unoconv \
-			gcc \
-			supervisor \
-			golang-go && \
-		go install github.com/HeavyHorst/unoconv-api && \
-        apt-get remove -y golang-go gcc && \
+			supervisor && \
+        apt-get remove -y && \
 	    apt-get autoremove -y && \
         apt-get clean && \
-	    rm -rf /var/lib/apt/lists/
+			rm -rf /var/lib/apt/lists/
 
 # Set the locale
 RUN locale-gen de_DE.UTF-8  

--- a/supervisord.conf
+++ b/supervisord.conf
@@ -8,7 +8,7 @@ autostart=true
 autorestart=true
 
 [program:unoconv-api]
-command=/go/bin/unoconv-api
+command=/opt/unoconv-api/unoconv-api
 stderr_logfile=/dev/stderr
 stderr_logfile_maxbytes=0
 stdout_logfile=/dev/stdout

